### PR TITLE
Set redcap repeat instance when writing linelist to cases

### PR DESCRIPTION
**Pull request by ljwoodley**
Addresses issue #140

### CODE REVIEW

File: load_roomates_into_linelist.R [hunk 1]
```diff
@@ -27,8 +27,34 @@ if (length(read_case_data) > 2) {
...
+    mutate(redcap_repeat_instance = if_else(is.na(redcap_repeat_instance), 0, redcap_repeat_instance)) %>%
+    group_by(record_id) %>%
+    summarise(max_redcap_repeat_instance = max(redcap_repeat_instance))
+
+  if (nrow(existing_linelist_entries) > 0) {
+
     case_record_and_infection_area <- case_data_read %>%
-      distinct(record_id, case_infection_area)
+      #  if there are multiple cases per infection area choose a unique case for that infection area
+      distinct(case_infection_area, .keep_all = T) %>%
```
Comment (ljwoodley) [inline load_roomates_into_linelist.R:54]: If there are multiple cases per infection area then a unique case for that infection_area is chosen. Thus, if xyz building is infected and case a and case b both live in xyz and were both included in one iteration of read_case_data then the line-list entries will be generated from only one of the given cases. This ensures that the redcap_repeat_instance generated is unique to a case

Comment (pbchase) [reply load_roomates_into_linelist.R:54]: The `distinct` here causes trouble when you use `case_record_and_infection_area` to create `case_linelist_upload`. If there really were two cases in the same infection area, only one of them will get line list entries. 

As you are using `distinct(case_infection_area)` in the creation of `infection_areas_to_read_from_housing` just below,  you can safely take out this distinct command and still get the desired effect in making the filter.

```suggestion
```

Comment (ljwoodley) [reply load_roomates_into_linelist.R:54]: That was the intended purpose. Are you saying that both cases should get the same line list entries if there were two cases in the same infection area?

Comment (ljwoodley) [reply load_roomates_into_linelist.R:54]: Also `case_record_and_infection_area` contains the redcap_repeat_instance. The `case_infection_area` is used for the join. If we remove that distinct we'll have to determine another way to get existing line list entries.

Comment (pbchase) [reply load_roomates_into_linelist.R:54]: Without my change, If two cases share the same infection area, the first one gets the correct roommates on the linelist. The other case gets nothing. I got the desired result when I removed this distinct.

**Pull request merged**